e2e_tests/tests/command/test_run.py

```diff
@@ -522,6 +523,8 @@ def test_horovod_optimizer(
     image = conf.TF1_GPU_IMAGE
     if tf2:
         image = conf.TF2_GPU_IMAGE
+    if not tf2 and "CUDA" in os.environ and os.environ["CUDA"] == "11":
```

Comment:
I thought we hardcoded the tf1 images to use the cuda 11 image anyway?

So... if `CUDA == 11` I think that the tf2 bool would not make any difference to the tests, right?  Actually it seems like a bunch of duplicate tests so I'm not opposed to skipping them, but IIUC the message this is emitting is not really why we should skip this test.

Reply:
Yeah the idea behind pointing the TF1 image variable to the TF2-only CUDA 11 image was that there's a bunch of tests that just use the default, which ends up being the TF1, so it's an easy way to get a lot more tests running on that image. For most tests, it turns out they don't care. Some are parameterized and would try run against both, which is a little wasteful (and given a lot more time I'd love to go over every test and make sure we run only what we need), but this was the only case where it turned out to matter. If it tried using TF1, it would fail. But all the TF2 runs pass.

Reply:
Right, but how would *this* test fail?  It seems like where pytest.skip() is called, we'd be secretly using TF2 anyway.

Reply:
Something was trying to do set_session: https://gist.github.com/mackrorysd/a87eda55c65dda42f810051c2636cf36

I didn't dig much deeper than that. The TF2 variant of the test was obviously running against TF2 successfully, which was my main concern.